atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -1476,7 +1483,22 @@ CfDef getCfForTable(TableReference tableRef, byte[] rawMetadata, int gcGraceSeco
             long timestamp,
             ConsistencyLevel consistency,
             Supplier<ResultsExtractor<T>> resultsExtractor) {
-        RowGetter rowGetter = new RowGetter(clientPool, queryRunner, consistency, tableRef, ColumnFetchMode.FETCH_ALL);
+        SlicePredicate predicate = new SlicePredicate();
+        if (rangeRequest.getColumnNames().size() == 1) {
+            byte[] col = rangeRequest.getColumnNames().iterator().next();
+            ByteBuffer start = CassandraKeyValueServices.makeCompositeBuffer(col, timestamp - 1);
+            ByteBuffer end = CassandraKeyValueServices.makeCompositeBuffer(col, -1);
+            SliceRange slice = new SliceRange(start, end, false, 1);
+            predicate.setSlice_range(slice);
```

Comment:
@clockfort i'm struggling to grasp why this was never implemented this way before. there don't seem to be any edge cases here because we literally discard every value except the most recent anyway... but yet i feel like there must be some reason we'd make such an obviously inefficient query here

Reply:
@nziebart maybe it was a limitation in Cassandra 1.2.X? CLocks would know more concretely though.